projects/flow.groups-2D-random-walk/src/project.py

```diff
@@ -0,0 +1,152 @@
+import flow
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+# Helper functions
+def get_first_replica(jobs, first_index=1):
+    try:
+        job = next(filter(lambda job: job.sp.replica == first_index, jobs))
+    except StopIteration:
+        raise RuntimeError("First replica not found.")
+    return job
+
+
+def positions_generator(jobs):
+    for job in jobs:
+        with job.data:
+            yield job.data.positions
+
+
+def aggregate_cond_true(doc_attr):
+    """Create a condition which is true if all jobs have doc_attr as truthy."""
+
+    def true_for_all_docs(*jobs):
+        return all(job.doc.get(doc_attr, False) for job in jobs)
+
+    return true_for_all_docs
+
+
+class Project(flow.FlowProject):
+    """Create a workflow for simulating 2D Gaussian random walks."""
+
+    pass
+
+
+@Project.label
+def simulated(job):
+    return "positions" in job.data
+
+
+def all_simulated(*jobs):
+    return all(simulated(job) for job in jobs)
+
+
+# Create aggregator that combines all replicas with a single standard deviation
+std_aggregator = flow.aggregator.groupby("std")
+
+
+# Define all groups
+# -----------------
+plot = Project.make_group("aggregate-plot", aggregator=std_aggregator)
+analyze_and_plot = Project.make_group("post-processing", aggregator=std_aggregator)
+
+
+@Project.post(simulated)
+@Project.operation
+def simulate(job):
+    """Simulate a 2D random walk."""
+    # use job id as random seed
+    n_steps = job._project.doc.run_steps
+    generator = np.random.default_rng(int(job.id, 16))
+    # Generate all moves at once since they are independent
+    moves = generator.normal(job.sp.mean, job.sp.std, (n_steps, 2))
+    position = np.empty((n_steps + 1, 2), dtype=float)
+    position[0, :] = [0, 0]
+    # Store the results: cumsum aggregates all previous moves from the origin
+    position[1:] = np.cumsum(moves, axis=0)
+    job.data.positions = position
+
+
+@analyze_and_plot
+@std_aggregator
+@Project.pre(all_simulated)
+@Project.post(aggregate_cond_true("msd_analyzed"))
```

Comment:
Try this. (See comment above.)
```suggestion
@Project.post.true("msd_analyzed")
```